File: AchuBan-ECom.Models/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AchuBan_ECom.Models.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public string? Description { get; set; }

        [Required]
        public string ISBN { get; set; }

        [Required]
        public string Author { get; set; }

        [Required]
        [Display(Name = "Price for 1-50")]
        public decimal ListPrice { get; set; }

        [Required]
        [Display(Name = "Price for 50+")]
        public decimal Price50 { get; set; }

        [Required]
        [Display(Name = "Price for 100+")]
        public decimal Price100 { get; set; }

        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        public required Category category { get; set; }
    }
}

File: AchuBan-ECom/wwwroot/js/product.js
$(document).ready(function () {
    loadDataTable();
});
function loadDataTable() {
    $('#productsTable').DataTable({
        "ajax": {
            "url": "/Admin/Products/GetAll"
        },
        "columns": [
            {
                "data": "imageUrl",
                "render": function (data) {
                    // hide image cell when there's no image
                    if (!data) return "";
                    return `<img src="${data}" style="max-height:100px;" />`;
                }
            },
            { "data": "name" },
            { "data": "isbn" },
            { "data": "author" },
            { "data": "listPrice" },
            { "data": "price50" },
            { "data": "price100" },
            { "data": "category.name" },
            {
                "data": "id",
                "render": function (data) {
                    return `
                                <a href="/Admin/Products/Upsert/${data}" class="btn btn-primary btn-sm me-2">
                                    <i class="bi bi-pencil-square"></i> Edit
                                </a>
                                <a href="javascript:void(0)" onClick="deleteProduct('/Admin/Products/Delete/${data}')" class="btn btn-danger btn-sm">
                                    <i class="bi bi-trash"></i> Delete
                                </a>
                            `;
                }
            }
        ]
    });
};

function deleteProduct(url) {
    Swal.fire({
        title: "Are you sure?",
        text: "You won't be able to revert this!",
        icon: "warning",
        showCancelButton: true,
        confirmButtonColor: "#3085d6",
        cancelButtonColor: "#d33",
        confirmButtonText: "Yes, delete it!"
    }).then((result) => {
        if (result.isConfirmed) {
            // read antiforgery token rendered on the page
            var token = $('input[name="__RequestVerificationToken"]').val();

            $.ajax({
                url: url,
                type: "POST", // controller expects POST (ActionName("Delete") + [HttpPost])
                headers: {
                    "RequestVerificationToken": token
                },
                success: function () {
                    Swal.fire({
                        title: "Deleted!",
                        text: "Product has been deleted.",
                        icon: "success"
                    }).then(function () {
                        // reload datatable after user closes the success dialog
                        $('#productsTable').DataTable().ajax.reload();
                    });
                },
                error: function (xhr) {
                    Swal.fire({
                        title: "Error",
                        text: "Delete failed. " + (xhr.responseText || ""),
                        icon: "error"
                    });
                }
            });
        }
    });
}